.github/workflows/ci.yml

```diff
@@ -1,98 +1,29 @@
-      - name: Setup Bundler
-        run: |
-          gem install bundler
-          bundle config path vendor/bundle
-          bundle lock --update
-      - name: Cache gems
-        uses: actions/cache@v3
-        with:
-          path: vendor/bundle
-          key: ${{ runner.os }}-gems-${{ hashFiles('**/Gemfile.lock') }}
-          restore-keys: |
-            ${{ runner.os }}-gems-
-      - name: Setup
-        run: bundle install --jobs=3 --retry=3
-      - name: Run rubocop
-        run: bundle exec rubocop
+    uses: theforeman/actions/.github/workflows/rubocop.yml@v0
+    with:
+      command: bundle exec rubocop
+  
   test:
-    runs-on: ubuntu-latest
+    name: Ruby
     needs: rubocop
-    services:
-      postgres:
-        image: postgres:12.1
-        ports: ["5432:5432"]
-        options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5
-    strategy:
-      fail-fast: false
-      matrix:
-        foreman-core-branch:
-          - 3.5-stable
-          - 3.6-stable
-          - 3.7-stable
-          - develop
-        node-version: [12]
-        ruby-version: [2.7]
-    steps:
-      - name: Install dependencies
-        run: |
-          sudo apt-get update
-          sudo apt-get install build-essential libcurl4-openssl-dev libvirt-dev ruby-libvirt zlib1g-dev libpq-dev
-      - uses: actions/checkout@v3
-        with:
-          repository: theforeman/foreman
-          ref: ${{ matrix.foreman-core-branch }}
-      - uses: actions/checkout@v3
-        with:
-          path: foreman_omaha
-      - name: Setup Ruby
-        uses: ruby/setup-ruby@v1
-        with:
-          ruby-version: ${{ matrix.ruby-version }}
-      - name: Setup Node
-        uses: actions/setup-node@v3
-        with:
-          node-version:  ${{ matrix.node-version }}
-      - name: Setup Bundler
-        run: |
-          gem install bundler
-          bundle config path vendor/bundle
-          bundle config set without journald development console mysql2 sqlite
-          echo "gem 'foreman_omaha', path: './foreman_omaha'" > bundler.d/foreman_omaha.local.rb
-          bundle lock --update
-      - name: Cache gems
-        uses: actions/cache@v3
-        with:
-          path: vendor/bundle
-          key: ${{ runner.os }}-gems-${{ hashFiles('**/Gemfile.lock') }}
-          restore-keys: |
-            ${{ runner.os }}-gems-
-      - name: Setup Plugin
-        run: |
-          bundle install --jobs=3 --retry=3
-          bundle exec rake db:create
-          bundle exec rake db:migrate
-          npm install
-          bundle exec rake webpack:compile
-      - name: Run plugin tests
-        run: |
-          bundle exec rake test:foreman_omaha
-          bundle exec rake test TEST="test/unit/foreman/access_permissions_test.rb"
-      - name: Precompile plugin assets
-        run: bundle exec rake 'plugin:assets:precompile[foreman_omaha]'
-        env:
-          RAILS_ENV: production
```

Comment:
@evgeni do we have this step implementation in https://github.com/theforeman/actions/blob/v0/.github/workflows/foreman_plugin.yml, i don't quite understand what this step is for?

Reply:
Correct, we have it in https://github.com/theforeman/actions/blob/98d24a3e0f0c3c9aaa6bf38f5ea5357b8b9bfa0c/.github/workflows/foreman_plugin.yml#L83

The idea is that we ensure that the plugin assets can be compiled with *production* settings (the code paths are slightly different, so it's better to ensure both dev and prod compiles fine)